pilot/pkg/kube/inject/inject.go

```diff
@@ -555,21 +561,24 @@ func InjectionData(sidecarTemplate, valuesConfig, version string, deploymentMeta
 		"contains":            flippedContains,
 	}
 
-	var tmpl bytes.Buffer
-	temp := template.New("inject")
-	t, err := temp.Funcs(funcMap).Parse(sidecarTemplate)
-	if err != nil {
-		log.Infof("Failed to parse template: %v %v\n", err, sidecarTemplate)
-		return nil, "", err
+	// Need to use FuncMap and SidecarTemplateData context
+	funcMap["render"] = func(template string) string {
```

Comment:
nit: can you put this directly into funcmap when it is initialized?

Reply:
I don't think we can put the initialization into function because the function needs to use funcmap

Reply:
Ah right, forgot about that.